src/utils/self_test/self_test.c

```diff
@@ -1908,6 +1943,7 @@ int main(int argc, char *argv[])
 	if (all_params != NULL)
 		D_FREE(all_params);
 	d_log_fini();
+	dc_mgmt_fini();
```

Comment:
similarly here should do it only   if (use_daos_agent_vars) 

Also this should be done before d_log_fini() in case any of dc_mgmt_* code logs messages

Reply:
38f8e41f157ae7b000b47dd0bd1eba0641f39a35